ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/Update.scala

```diff
@@ -25,6 +25,12 @@ sealed trait Update extends Product with Serializable {
   def recordTime: Timestamp
 }
 
+/** An update describing metadata changes.
+  *
+  * Metadata changes regard parties, packages, ledger configurations and command rejections.
+  */
+sealed trait MetadataUpdate extends Update
```

Comment:
What's the purpose of this change?

Reply:
I have added it for making threading through the pipeline stages more type safe and modular (e.g. in the pipelined update flow, we either act on the `MetadataUpdate` or on the `TransactionAccepted`).

Reply:
This seems to split ledger updates to "transaction accepted" and "all other changes". This is related to a particular indexer optimization, and I'm not sure such a trait belongs to the participant state interface.

Before this PR, we had:
```
OffsetUpdate = PreparedTransactionInsert | OffsetStepUpdatePair
```
where `OffsetStepUpdatePair` could include any update (including transactions). It looks like you added `MetadataUpdate` so that you have:
```
OffsetUpdate = (PreparedTransactionInsert | MetadataUpdateStep) | OffsetStepUpdatePair
```
where `MetadataUpdateStep` is the same as `OffsetStepUpdatePair`, except that it's guaranteed to not include transactions. Do we really need that split? If anything, shouldn't it be:
```
OffsetUpdate = PreparedTransactionInsert | UnpreparedUpdateThatIsAnythingButATransactionInsert
```
?

The main benefit seems to be to avoid duplicating the code that handles transaction updates. Before this PR, we had duplicated code and a logger warning for this: https://github.com/digital-asset/daml/pull/8571/files#diff-63690f91ba83940aae6dc820d83d141caec268ff769438437ee49703f3a9d238L391.

Reply:
As it is a more widely-used API, I've removed the change.